Description: render FIREHOL_LOAD_KERNEL_MODULES=0 more effective
 If set to 0, the variable FIREHOL_LOAD_KERNEL_MODULES (firehol-variables(5))
 is rendered to no more emit the `IMPORTANT WARNING' related to the absence
 (non finding) of current kernel configuration files.
Origin: debian
Author: Jerome Benoit <calculus@rezozer.net>
Last-Update: 2014-08-06

--- a/sbin/firehol.in
+++ b/sbin/firehol.in
@@ -4036,19 +4036,24 @@
 	# We could not find a kernel configuration
 	
 	KERNEL_CONFIG=
-	echo >&2 " "
-	echo >&2 " IMPORTANT WARNING:"
-	echo >&2 " ------------------"
-	echo >&2 " FireHOL cannot find your current kernel configuration."
-	echo >&2 " Please, either compile your kernel with /proc/config,"
-	echo >&2 " or make sure there is a valid kernel config in:"
-	echo >&2 " /usr/src/linux/.config"
-	echo >&2 " "
-	echo >&2 " Because of this, FireHOL will simply attempt to load"
-	echo >&2 " all kernel modules for the services used, without"
-	echo >&2 " being able to detect failures."
-	echo >&2 " "
-	sleep 2
+
+	if [ ! ${FIREHOL_LOAD_KERNEL_MODULES} -eq 0 ]
+	then
+		echo >&2 " "
+		echo >&2 " IMPORTANT WARNING:"
+		echo >&2 " ------------------"
+		echo >&2 " FireHOL cannot find your current kernel configuration."
+		echo >&2 " Please, either compile your kernel with /proc/config,"
+		echo >&2 " or make sure there is a valid kernel config in:"
+		echo >&2 " /usr/src/linux/.config"
+		echo >&2 " "
+		echo >&2 " Because of this, FireHOL will simply attempt to load"
+		echo >&2 " all kernel modules for the services used, without"
+		echo >&2 " being able to detect failures."
+		echo >&2 " "
+		sleep 2
+	fi
+
 fi
 
 # activation-phase command to check for the existance of
